src/cockpit/protocol.py

```diff
@@ -40,10 +38,11 @@ class CockpitProblem(Exception):
     It is usually thrown in response to some violation of expected protocol
     when parsing messages, connecting to a peer, or opening a channel.
     """
-    def __init__(self, problem: str, **kwargs):
-        super().__init__(kwargs.get('message') or problem)
+    def __init__(self, problem: str, _msg: 'JsonObject | None' = None, **kwargs: JsonDocument) -> None:
```

Comment:
This was intended.  attr is the complete sum of `problem`, `_msg` and `kwargs` — not just `kwargs`.  It's all the attributes, together, that will get sent as part of the message, except for `command="close"` or `command="init"`, which will get added depending on context.

Reply:
We can't do that because we need to send these using the normal mechanism for sending control messages (and also add `command="close"` or `command="init"` as appropriate).

I'm willing to discuss this further, but I really like the way this works now.

One thing I've done is to remove the redundant `problem` field.  I don't think anything uses it, since the problem is now part of the `attrs` dict.

Reply:
Ack, thanks. This certainly isn't a blocker, I just wanted to understand the background